vcs/gitcmd/repo.go

```diff
@@ -411,23 +414,25 @@ func (r *Repository) commitLog(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, err
 	}
 
 	// Count commits.
-	cmd = exec.Command("git", "rev-list", "--count", rng)
-	if opt.Path != "" {
-		// This doesn't include --follow flag because rev-list doesn't support it, so the number may be slightly off.
-		cmd.Args = append(cmd.Args, "--", opt.Path)
-	}
-	cmd.Dir = r.Dir
-	out, err = cmd.CombinedOutput()
-	if err != nil {
-		return nil, 0, fmt.Errorf("exec `git rev-list --count` failed: %s. Output was:\n\n%s", err, out)
-	}
-	out = bytes.TrimSpace(out)
-	total, err := strconv.ParseUint(string(out), 10, 64)
-	if err != nil {
-		return nil, 0, err
+	if total != nil {
```

Comment:
You're looking at code inside `commitLog`. It's unexported, and only has two internal "users". One use on line 296 and the other one line 326.

The public API, and behavior of this library remains completely unchanged.

> If a user doesn't care about the information they will leave this as null?

Yes, see the documentation change on line 339.